packages/nimble-components/src/table/types.ts

```diff
@@ -40,6 +40,18 @@ export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
 
+/**
+ * InternalTableRecord describes a hierarchical data structure that is used for
+ * the internal representation of the data, and allows us to represent data with
+ * parent-child relationships within Tanstack.
+ */
+export interface InternalTableRecord<
+    TRecord extends TableRecord = TableRecord
+> {
+    subRows?: InternalTableRecord<TRecord>[];
+    data: TRecord;
```

Comment:
how would you feel about calling this `originalRecord`, `userRecord`, `providedRecord`?
I think I like `userRecord` the most. It's a lot clearer to me what the context is when reading a test like the following:
```ts
 const data = [
      { data: { stringField: 'value-1', numberField: 10 } },
      { data: { stringField: 'value-2', numberField: 11 } }
  ];

// vs

 const data = [
    { userRecord: { stringField: 'value-1', numberField: 10 } },
    { userRecord: { stringField: 'value-2', numberField: 11 } }
];
```
Also could we do some typing around saying it is readonly?

Reply:
I like the idea of a more descriptive name but have a minor preference not to use the term "user" since I'm on a fruitless crusade to use language that differentiates between end-users and clients. I'd vote `originalRecord` or `clientRecord`.

Reply:
Think in this context I'd lean more to `clientRecord`. Seems more indicative of the thing that came from outside than `originalRecord`